miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -322,15 +322,24 @@ HotTarget.sample = (function() {
                                     return sam.alias + " -- " + sam.externalName;
                                   });
                             }
-                            var hasIdentityInProject = (identitiesSources.length > 0 
-                                && data.matchingIdentities[0].projectId == selectedProject.id
-                                && data.matchingIdentities[0].externalName == flat.externalName);
-                            if (!hasIdentityInProject) {
+                            var indexOfMatchingIdentityInProject = function () {
```

Comment:
You can use `findIndex` https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex